### DISCUSSION TITLE
Help with listening for LocalChatSpyEvents

### DISCUSSION THREAD

**Discussion by networkException** [EssentialsX/Essentials#3782] [Q&A] [answered]
Hello there,

I'm trying to work on a plugin that utilizes the `LocalChatSpyEvent` as a previous solution using the `PlayerCommandPreprocessEvent` and regex is not reliable enough. 

I tried registering an event handler like I would do for any other event but it's sadly not triggering.
```java
public class SomeListener implements Listener {
    @EventHandler
    public void onChatSpy(LocalChatSpyEvent event) {
        System.out.println(event.getEventName());
    }
}

// in onEnable
getServer().getPluginManager().registerEvents(new SomeListener(), this);
```

Is there anything special I need to consider?

**Comment by triagonal** [comment, marked as answer]
I was just able to successfully listen to and interact with this event. Your code looks fine, but there's two things you need to make sure you're doing:

1. You need to have EssentialsXChat installed, since chat is the plugin that fires the `LocalChatSpyEvent`.
2. The chat needs to actually be local, i.e. you need to set `chat.radius` in the chat section of the config to something greater than 0.

Keep in mind that this event is only for local chat messages and won't fire in certain situations like shouting (`!message`) or questions (`?message`).

**Reply by networkException** [reply]
Thank you a lot for the answer, turns out that I was just lacking understanding of what the event really does
